core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java

```diff
@@ -545,7 +545,7 @@ private static RubyBigDecimal newInstance(Ruby runtime, IRubyObject recv, RubyFl
         RubyBigDecimal res = newFloatSpecialCases(runtime, arg);
         if (res != null) return res;
 
-        return new RubyBigDecimal(runtime, (RubyClass) recv, new BigDecimal(arg.getDoubleValue(), mathContext));
+        return new RubyBigDecimal(runtime, (RubyClass) recv, new BigDecimal(arg.inspect().asJavaString(), mathContext));
```

Comment:
I believe this can just be `arg.toString()`, avoiding the intermediate RubyString object.

Reply:
@headius thanks